src/nupic/algorithms/Cells4.hpp

```diff
@@ -416,11 +417,12 @@ namespace nupic {
          * This also called when lifting from persistence.
          */
         void
-        initialize(UInt nColumns =0, UInt nCellsPerCol =0,
-                   UInt activationThreshold =1,
-                   UInt minThreshold =1,
-                   UInt newSynapseCount =1,
-                   UInt segUpdateValidDuration =1,
+        initialize(UInt nColumns =500, 
```

Comment:
I think setting defaults for parameters like `nColumns` and `nCellsPerCol` doesn't make sense. The default won't just be bad, it won't even work. The other parameters might be bad, but at least it will still work. So I agree with the previous defaults of `0` for the first two parameters which effectively makes them required arguments.